File: sum_two/sum_two.rb
def sum_two(array, target)
    hash = {}

    array.each_with_index do |n, i|
        diff = target - n
        if hash.keys.include?(diff)
            return [hash[diff], i]
        end
      hash[n] = i
    end
    nil
end

array = [2,11,7,15]
target = 9

puts sum_two(array, target)

File: sum_two/sum_two.php
<?php


function sumTwo($array, $target){
    $hash = [];

    foreach ($array as $index => $value){
        $diff = $target - $value;
        if(array_key_exists($diff, $hash)){
            return [$hash[$diff],$index];
        }
        $hash[$value] = $index;
    }

    return null;
}
$hash = [];
$array = [2,7,11,15];
$target = 9;
print_r(sumTwo($array, $target));


// print array
// echo implode(', ',$array);

// print int 
//echo var_dump($target);
